File: php/features.php
<?php
declare(strict_types=1);

// PokemonTcg SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PokemonTcgFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PokemonTcgBaseFeature();
            case "test":
                return new PokemonTcgTestFeature();
            default:
                return new PokemonTcgBaseFeature();
        }
    }
}

File: py/features.py
# PokemonTcg SDK feature factory

from feature.base_feature import PokemonTcgBaseFeature
from feature.test_feature import PokemonTcgTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PokemonTcgBaseFeature(),
        "test": lambda: PokemonTcgTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
